omniqueue/src/backends/redis/mod.rs

```diff
@@ -1,15 +1,44 @@
+//! Redis stream-based queue implementation
+//!
+//! # Redis Streams in Brief
+//! Redis has a built-in queue called streams. With consumer groups and consumers, messages in this
+//! queue will automatically be put into a pending queue when read and deleted when acknowledged.
+//!
+//! # The Implementation
+//! This implementation uses this to allow worker instances to race for messages to dispatch which
+//! are then, ideally, acknowledged. If a message is processing for more than 45 seconds, it is
+//! reinserted at the back of the queue to be tried again.
+//!
+//! This implementation uses the following data structures:
+//! - A "tasks to be processed" stream - which is what the consumer listens to for tasks.
+//!     AKA: Main
+//! - A ZSET for delayed tasks with the sort order being the time-to-be-delivered
+//!     AKA: Delayed
+//!
+//! The implementation spawns an additional worker that monitors both the zset delayed tasks and
+//! the tasks currently processing. It monitors the zset task set for tasks that should be
+//! processed now, and the currently processing queue for tasks that have timed out and should be
+//! put back on the main queue.
```

Comment:
I may need to rewrite this copy. I also need to re-read the "pending" code in OSS since I initially thought it was "legacy stuff" but now that I see this copypasta, I'm not sure.

Leaving this comment here to make sure I resolve this before merging.

Reply:
After re-reading [`background_task_pending`](https://github.com/svix/svix-webhooks/blob/4eb0fb9c61709ac60ad829ba06ab22832f57e08e/server/svix-server/src/queue/redis.rs#L130) in `svix-server`, my sense is this is necessary, but orthogonal to the delayed message support.

I'm not sure why it wasn't included in the initial dev effort for redis support _other than_ it's weird to be doing all this in userland and it didn't directly serve the `svix-bridge` effort.

At any rate, it seems like we need it if we want to serve `svix-server`.

I'll get working on adding this in a 2nd rev in case we decide we want to split it into a 2nd PR.

Reply:
I went ahead and put this as a separate PR since I'll have to add a fresh set of tests to even know if this works how I think it does.

https://github.com/svix/omniqueue-rs/pull/20